### ISSUE TITLE
Product features

### ISSUE THREAD

**Issue by kazuhiroinaba** [CS2103AUG2016-W10-C1/main#28]
## Must have features
- [x] Supports 3 item types (i.e. events that has a start time and end time, tasks that have to be done before a specific deadline, tasks without specific times)
- [x] CRUD function for items (i.e. marking an item as done)
- [x] Undo
- [x] Flexibility in command format (support a few natural variations of the command format)
- [x] Search items with keywords
- [x] Keeps track of items that are done or not done
- [x] Specify a folder as the data storage location
## Nice to have features
- Support recurring tasks
- More powerful search (i.e. search for empty slots, near-match search, auto-complete, filter with other attributes)
- More flexible commands (i.e. non-strict ordering of keywords, ability to specify aliases for commands, support for more ‘natural’ language input, multiple undo/redo)
- Good GUI (i.e. in-built guidance for new users, good visual feedback for actions, feedback while typing commands, hotkey to activate/hide, notifications, more important/urgent tasks stand out from the rest, longer events look different from shorter events, easy to see free slots)
- Google integration (i.e. Google Calendar integration such as upload todo items to GCal, two-way sync, support for GCal quick add command format, etc. You may also use the Google Tasks API for this)
- Automated testing (achieve a very high level of automated testing)
- Launch application with keyboard shortcut (i.e. Alt + Space)
- "Block” multiple slots when the exact timing of a task is uncertain, and release the blocked slots when the time is finalized. (i.e. Block Mon 2-3pm and Tue 2-3pm for a meeting with boss → the meeting is confirmed for Tue → automatically release Mon slot)
- Postponing an item (i.e. overdue, unable to complete)
- **You can propose your own (needs approval)**

**Comment by jia1** [comment]
What's the update?
